packages/react/src/components/form/Checkbox/Checkbox.tsx

```diff
@@ -5,90 +5,52 @@ import { forwardRef } from 'react';
 
 import { omit } from '../../../utilities';
 import { Label } from '../../Label';
-import { Paragraph } from '../../Paragraph';
+import { ValidationMessage } from '../../ValidationMessage';
+import { Field } from '../Field';
+import { Input } from '../Input';
 import type { FormFieldProps } from '../useFormField';
 
 import { useCheckbox } from './useCheckbox';
 
 export type CheckboxProps = {
   /** Checkbox label */
-  children?: ReactNode;
+  label?: ReactNode;
+  /** Description for field */
+  description?: ReactNode;
   /** Value of the `input` element */
   value: string;
+  /** Validation message for field */
+  validation?: ReactNode;
+  /**
+   * Changes field size and paddings
+   * @default md
+   */
+  size?: 'sm' | 'md' | 'lg';
   /**Toggle indeterminate state for Checkbox
    * @default false
    */
   indeterminate?: boolean;
-} & Omit<FormFieldProps, 'error' | 'errorId'> &
-  Omit<InputHTMLAttributes<HTMLInputElement>, 'size' | 'value'>;
+} & Omit<InputHTMLAttributes<HTMLInputElement>, 'size'>;
 
 export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
-  (props, ref) => {
-    const { children, description, className, style, ...rest } = props;
-    const {
-      inputProps,
-      descriptionId,
-      hasError,
-      size = 'md',
-      readOnly,
-    } = useCheckbox(props);
-
-    const inputRef = useMergeRefs<HTMLInputElement>([
-      ref,
-      (el) => {
-        if (el) {
-          el.indeterminate = rest.indeterminate ?? false;
-        }
-      },
-    ]);
+  function Checkbox(
+    { children, label, description, validation, ...rest },
+    ref,
+  ) {
+    // const inputRef = useMergeRefs<HTMLInputElement>([
+    //   ref,
+    //   (el) => {
+    //     if (el) el.indeterminate = rest.indeterminate ?? false;
+    //   },
+    // ]);
```

Comment:
Should this be in input instead? 🤔 Or simply remove, and instead show in a story, as this prevents server component behavior?

Reply:
Or move to hook?